### PR TITLE
Analyze generated code for CA2100 and CA3147 security rules

### LINKED ISSUE #4744: Consider enabling CA2100 (security analyzer) against generated code
### Analyzer

**Diagnostic ID**: [CA2100](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2100)

Consider enabling CA2100 against generated code. It's currently not enabled:

https://github.com/dotnet/roslyn-analyzers/blob/b148fa89c974a72c28184931767835709f34d1fa/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Data/ReviewSqlQueriesForSecurityVulnerabilities.cs#L41

Also consider moving it from `Data` directory to `Security` directory.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #4744

### CODE REVIEW

**Review by Evangelink — APPROVED** [review]
LGTM

@sharwell do we have any way to add a test for this so we can prevent regression?

**Comment by sharwell** [comment]
Yes, just write a test with `// <auto-generated/>` as the first line.

**Comment by Evangelink** [comment]
@Youssef1313 would it be possible to add tests with the comment suggested by @sharwell ?

**Comment by Youssef1313** [comment]
@Evangelink This could apply to *nearly* all security analyzers. Is there a way to have miscellaneous tests for tasks like this?

**Comment by Evangelink** [comment]
@Youssef1313 I don't think so. I would rule in favor of adding (or updating) only one test for `CA2100` and `CA3147` as this is the scope of this PR and leave the other analyzers out.

**Review by Evangelink — APPROVED** [review]
Thanks @Youssef1313

**Comment by mavasani** [comment]
Tagging @dotpaul - would you prefer the PR target 2.9.x or you are fine if the change goes in master?

**Comment by dotpaul** [comment]
2.9.x please

File: src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/MarkVerbHandlersWithValidateAntiforgeryTokenAnalyzer.cs [hunk 1]
```diff
@@ -97,8 +97,8 @@ public partial class MarkVerbHandlersWithValidateAntiforgeryTokenAnalyzer : Diag
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            analysisContext.EnableConcurrentExecution();
-            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
```
Comment (Youssef1313) [inline src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/MarkVerbHandlersWithValidateAntiforgeryTokenAnalyzer.cs:101]:
```suggestion
            analysisContext.EnableConcurrentExecution();
            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
```

**Review by dotpaul — APPROVED** [review]
Thank you, @Youssef1313 !

**Pull request merged**